src/System.Private.CoreLib/shared/System/Double.cs

```diff
@@ -217,47 +246,47 @@ public unsafe override int GetHashCode()
         public override String ToString()
         {
             Contract.Ensures(Contract.Result<String>() != null);
-            return FormatProvider.FormatDouble(m_value, null, null);
+            return Number.FormatDouble(m_value, null, NumberFormatInfo.CurrentInfo);
```

Comment:
`class Number` is enclosed by the FormatProvider that we are getting rid of: https://github.com/dotnet/corert/blob/master/src/System.Private.CoreLib/src/System/Globalization/FormatProvider.FormatAndParse.cs#L12 . You can try removing the FormatProvider enclosing type to make it match CoreCLR. I am not sure how many other issues it is going to uncover. If it is not converging, ifdef the problematic globalization places in the shared code `#if CORECLR <the CoreCLR code> #else <the CoreRT code> #endif` and we will clean it up later.